binary-com/deriv-com
# Cypress end-to-end testing setup

# Summary

This pull request sets up Cypress for end-to-end testing of our staging/local environment along with a sample test for signup flow

## Changes

- Installed Cypress as a dev dependency
- Added `cypress:open` script to package.json
- Created first test for signup flow at `/cypress/e2e/signup.cy.ts` for both eu and non-eu website
- Created second test for footer exists at `/cypress/e2e/footer.cy.ts` for both eu and non-eu website
- Create a cypress.env.json file in the root and update the file with your base url otherwise the the test will only run in staging url

```
{
      "base_url": "http://localhost:8000"
} 
```


```js
describe('Signup page is functional for staging', () => {
  it('allows user to signup', () => {
    let url = Cypress.env('base_url') || 'https://staging.deriv.com';
    cy.visit(`${url}/signup`)
    cy.get('input#email_address').type('test@example.com')
    cy.get('input#terms_and_conditions').click()
    cy.contains('button', 'Create demo account').click()
    cy.url().should('include', '/signup-success?email=test@example.com') 
  })
})

## Testing
- Run npm run cypress:open to open the Cypress Test Runner
- Execute the sample signup test to verify it passes against staging
## Next Steps
- Add more end-to-end tests covering critical user flows
- Configure CI to run Cypress tests on every commit

Let me know if you have any other questions!
